addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/service/MainTVServerService.java

```diff
@@ -176,7 +176,7 @@ public void onServiceSubscribed(String service, boolean succeeded) {
     public void onValueReceived(String variable, String value, String service) {
 
         String oldValue = stateMap.get(variable);
-        if (value.equals(oldValue)) {
+        if (oldValue.equals(value)) {
```

Comment:
No in the original case the binding would fall apart with an NPE, but I can add it, but I would prefer to add it like this, its the same but more explicit:

```java
if ((value == null && oldValue == null) || (value != null && value.equals(oldValue))) {
    logger.trace("Value '{}' for {} hasn't changed, ignoring update", value, variable);
    return;
}
```